.github/workflows/test.yml.jinja

```
@@ -0,0 +1,65 @@
+name: test Odoo addons
+
+on:
+  pull_request:
+    branches:
+      - "{{ odoo_version }}*"
+  push:
+    branches:
+      - "{{ odoo_version }}*"
+
+{%
+set IMAGES = {
+  "odoo": {
+    "14.0": "ghcr.io/oca/oca-ci/py3.6-odoo14.0:latest",
+    "13.0": "ghcr.io/oca/oca-ci/py3.6-odoo13.0:latest",
+    "12.0": "ghcr.io/oca/oca-ci/py3.6-odoo12.0:latest",
+    "11.0": "ghcr.io/oca/oca-ci/py3.5-odoo11.0:latest",
+    "10.0": "ghcr.io/oca/oca-ci/py2.7-odoo10.0:latest",
+  },
+  "ocb": {
+    "14.0": "ghcr.io/oca/oca-ci/py3.6-ocb14.0:latest",
+    "13.0": "ghcr.io/oca/oca-ci/py3.6-ocb13.0:latest",
+    "12.0": "ghcr.io/oca/oca-ci/py3.6-ocb12.0:latest",
+    "11.0": "ghcr.io/oca/oca-ci/py3.5-ocb11.0:latest",
+    "10.0": "ghcr.io/oca/oca-ci/py2.7-ocb10.0:latest",
+  }
+}
+-%}
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: {{ "${{" }} matrix.container {{ "}}" }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - container: {{ IMAGES["odoo"][odoo_version] }}
+            makepot: "true"
+          - container: {{ IMAGES["ocb"][odoo_version] }}
+    services:
+      postgres:
+        image: postgres:9.6
+        env:
+          POSTGRES_USER: odoo
+          POSTGRES_PASSWORD: odoo
+          POSTGRES_DB: odoo
+        ports:
+          - 5432:5432
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - run: oca_install_addons
+      - run: oca_init_test_database
+      - name: licenses check
+        run: manifestoo -d . check-licenses
+      - name: development status check
+        run: manifestoo -d . check-dev-status --default-dev-status=Beta
+      - run: oca_run_tests
+      - uses: codecov/codecov-action@v1
+      - run: |
+          oca_export_and_commit_pot
+          git fetch --unshallow --all
+          git push https://x-access-token:{{ "${{" }} secrets.GIT_PUSH_TOKEN {{ "}}" }}@github.com/{{ "${{" }} github.repository {{ "}}" }}
+        if: {{ "${{" }} matrix.makepot == 'true' && github.event_name == 'push' && github.repository_owner == 'OCA' && startsWith(github.ref, 'refs/heads/{{ odoo_version }}') {{ "}}" }}
```

What do you think of using something like https://github.com/marketplace/actions/github-commit-push to simplify here? The Github Actions from the marketplace are a great way of simplifying and reduce maintenance long term...

Also, the syntax could be a little simplified using [Jinja's `raw` block](https://jinja.palletsprojects.com/en/2.11.x/templates/#escaping). The only Jinja variable used in the if expression is `odoo_version`, but even that variable could be made part of the job variables or, as explained next, even that part of the expression could potentially be removed.

As the job will only run for pull requests and pushes to the main branch, we can simply check if we are in a push event (which means we are necessarily in a push event in the main branch)

```suggestion
      - run: oca_export_and_commit_pot
      - name: Commit & Push changes
        {% raw %}
        if: ${{ matrix.makepot == 'true' && github.event_name == 'push' && github.repository_owner == 'OCA' }}
        {% endraw %} 
        uses: actions-js/push@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
```

---

Thanks for the tip with the raw block. I'll test it.
Regarding the push action, I think I'll keep it explicit because it's just 2 lines of quite simple code and I don't think its worth pulling a dependency for that, especially a third party one.

---

done